tests/rules/javascript/test.shallow_wrapper.js

```diff
@@ -0,0 +1,62 @@
+import { VALIDATION_WARNING } from 'const';
+import JavaScriptScanner from 'scanners/javascript';
+import * as messages from 'messages';
+
+
+describe('shallow_wrapper', () => {
+
+  it('should allow use of XPCNativeWrapper with only one argument', () => {
+    var code = `XPCNativeWrapper(unsafeWindow);`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 0);
+      });
+  });
+
+  it('should not allow use of shallow wrapper with a variable', () => {
+    var code = `XPCNativeWrapper(unsafeWindow, 'foo');`;
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    return jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].code, messages.SHALLOW_WRAPPER.code);
+        assert.equal(validationMessages[0].type, VALIDATION_WARNING);
+
+        code = `var wrap = new XPCNativeWrapper(unsafeWindow, 'doc', 'a');`;
+        jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+        return jsScanner.scan();
+      })
+      .then((validationMessages) => {
```

Comment:
You could roll these up into a single bit of code to test and then check that you get 2 errors and then iterate over the validationMessages and check the code is as expected in each case.